crates/sdk/src/commit.rs

```diff
@@ -108,3 +111,39 @@ pub fn commit_app_exe(
     let app_engine = BabyBearPoseidon2Engine::new(app_fri_params);
     Arc::new(VmCommittedExe::<SC>::commit(exe, app_engine.config.pcs()))
 }
+
+/// Byte representation of AppExecutionCommit. Because the commits in AppExecutionCommit
+/// are stored as an array representing a base-F number, the 256 bits that make up each
+/// commit are different than actual the 256-bit representation of the Bn254. This class
+/// stores the latter to enable consistent serialization between different proof outputs.
+#[serde_as]
+#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
+pub struct AppExecutionCommitBytes {
+    #[serde_as(as = "serde_with::hex::Hex")]
+    /// 1 Bn254Fr public value for app exe commit in big-endian bytes.
+    pub app_exe_commit: [u8; BN254_BYTES],
+    #[serde_as(as = "serde_with::hex::Hex")]
+    /// 1 Bn254Fr public value for app vm commit in big-endian bytes.
+    pub app_vm_commit: [u8; BN254_BYTES],
```

Comment:
I'd prefer is `[u8; BN254_BYTES]` is some protected newtype that's opaque (so user just seems it's some Commit type) and then we convert, as you've done, to Bn254Fr, [F;8], etc etc